### PR TITLE
Git cola 4.4.1

### PR DESCRIPTION
**Pull request by bearsh**
Bug: https://bugs.gentoo.org/920572
Bug: https://bugs.gentoo.org/920534

### CODE REVIEW

**Comment by parona-source** [comment]
> I've created a pull request (https://github.com/gentoo/gentoo/pull/34445) for 4.4.1 which build fine and removed 4.3.1 which does not.

> I'm not sure what to do for the stable 4.2.1 version though

Seeing the 4.2.1 wont work at all due to setuptools-69 being stable and the only version available, I think removing broken version to get stable users to downgrade to the working 4.1.0 would be reasonable.

But any gentoo-dev who is merging should give some input on that.

**Comment by bearsh** [comment]
another way would be patching 4.2.1 with (https://github.com/git-cola/git-cola/commit/b53983c7fa7d29dd6f474177015fcb6f2ee0e6d1) until 4.4.1 hits stable.

**Comment by bearsh** [comment]
@arthurzam, as you committed the last bump, any suggestion on how to handle the stable 4.2.1 version?

File: dev-vcs/git-cola/git-cola-4.2.1-r1.ebuild [hunk 1]
```diff
@@ -38,6 +38,7 @@ BDEPEND="sys-devel/gettext
 		)
 	")
 "
+PATCHES=( "${FILESDIR}/pyproject.toml.patch" )
```
Comment (parona-source) [inline dev-vcs/git-cola/git-cola-4.2.1-r1.ebuild:41]: Thats not a great patch name.

Convention is ${P}-<what it does>.patch

So something like:
git-cola-4.2.1-fix-setuptools.patch

Comment (parona-source) [reply dev-vcs/git-cola/git-cola-4.2.1-r1.ebuild:41]: Also include links to the bug and commit in git-cola to the patch header.

**Comment by parona-source** [comment]
You could address the missing setuptools-scm bdep in the 4.4.1 bump. Python3.12 could be a nice bonus.

 PythonMissingSCMDependency
defines SETUPTOOLS_SCM_PRETEND_VERSION but is missing BDEPEND on setuptools-scm or alike
PythonCompatUpdate
PYTHON_COMPAT update available: python3_12

**Comment by thesamesam** [comment]
Thanks! Merged with some squashing for the py3.12 commit.

**Comment by bearsh** [comment]
wow, this was fast!!!

**Pull request closed without merging**